purchases/src/main/kotlin/com/revenuecat/purchases/Backend.kt

```diff
@@ -126,9 +137,10 @@ internal class Backend(
             "presented_offering_identifier" to offeringIdentifier,
             "observer_mode" to observerMode,
             "price" to price,
-            "currency" to currency
-        ).mapNotNull { entry: Map.Entry<String, Any?> ->
-            entry.value?.let { entry.key to it }
+            "currency" to currency,
+            "attributes" to subscriberAttributes.takeUnless { it.isEmpty() }?.toBackendMap()
+        ).mapNotNull { (key, value) ->
+            value?.let { key to it }
```

Comment:
maybe keeping this `mapNotNull` as a one-liner is easier to read? 
```kotlin
val body = mapOf(
        ...
        ).mapNotNull { (key, value) -> value?.let { key to it }
        }.toMap()
```

Reply:
could `mapNotNull` be replaced by `filterNotNull()`? https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/filter-not-null.html

Reply:
oh, maybe `filterNotNull` is list-only. but `filter` and `filterNot` might work. Using one of those might remove the need for the `toMap()`?

Reply:
Great idea